### PR TITLE
Created simple version of inline keyboard message and many changes and improves of exists code.

### PR DESCRIPTION
**Pull request by rusboard**

### CODE REVIEW

**Comment by rusboard** [comment]
Please correct Travis for CustomKeyboard folder.
There is 2 *.ino files now and one file was renamed.

**Comment by witnessmenow** [comment]
Awesome work! Inline keyboard was on my to do list so great to get it in there.

I think for the examples to work in the arduino ide It needs to be in the format name/name.ino could you check it works in the current format?

I'll take a proper look at the changes tonight or tomorrow when I'm not on my phone

**Comment by rusboard** [comment]
Yes, you are right about format for Arduino IDE, thx, I fixed this.

And I understood how to modify Travis "All checks have passed", yay! :)

File: examples/ESP8266/CustomKeyboard/InlineKeyboardMarkup/InlineKeyboardMarkup.ino [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+long Bot_lasttime;   //last time messages' scan has been done
+
+void handleNewMessages(int numNewMessages) {
+  Serial.println("handleNewMessages");
+  Serial.println(String(numNewMessages));
+
+  for (int i=0; i<numNewMessages; i++) {
+    String chat_id = String(bot.messages[i].chat_id);
+    String text = bot.messages[i].text;
+    String from_name = bot.messages[i].from_name;
```
Comment (witnessmenow) [inline examples/ESP8266/CustomKeyboard/InlineKeyboardMarkup/InlineKeyboardMarkup.ino]: A check for a blank name is in the other example, maybe its needed here too?

**Comment by rusboard** [comment]
Sure, tomorrow will do.

10 янв. 2017 г. 0:41 пользователь "Brian Lough" <notifications@github.com>
написал:

**Comment by rusboard** [comment]
Added from_name checking.

**Pull request merged**